app/models/comment.php

```diff
@@ -75,4 +140,214 @@ public function edit($comment_id)
+                'disliked' => 1
+            );
+
+            $insert = $db->insert('like_monitor', $params);
+            $update = $this->updateDislikedCount($comment_id);
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }    
+
+    //increments liked column in comment table
+    public function updateLikedCount($comment_id)
+    {
+        $comment = new Comment();
+       
+        try {
+            $db = DB::conn();
+            $db->begin();
+
+            $update = $db->query('UPDATE comment SET liked = liked + 1 WHERE id = ?', array($comment_id));
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    //increments disliked column in comment table
+    public function updateDislikedCount($comment_id)
+    {
+        $comment = new Comment();
+
+        try {
+            $db = DB::conn();
+            $db->begin();
+
+            $update = $db->query('UPDATE comment SET disliked = disliked + 1 WHERE id = ?', array($comment_id));
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    //decrements liked column in comment table
+    public function subtractLikedCount($comment_id)
+    {
+        $comment = new Comment();
+
+        try {
+            $db = DB::conn();
+            $db->begin();
+        
+            $update = $db->query('UPDATE comment SET liked = liked - 1 WHERE id = ?', array($comment_id));
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    //decrements disliked column in comment table
+    public function subtractDislikedCount($comment_id)
+    {
+        $comment = new Comment();
+        
+        try {
+            $db = DB::conn();
+            $db->begin();
+
+            $update = $db->query('UPDATE comment SET disliked = disliked - 1 WHERE id = ?', array($comment_id));
+
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    public function count($user_id)
+    {
+        $db = DB::conn();
+
+        $count_comments = $db->value('SELECT COUNT(id) FROM comment WHERE user_id = ?', array($user_id));
+
+        return $count_comments;
+    }
+
+    public function countLike($user_id)
+    {
+        $db = DB::conn();
+
+        $count_likes = $db->value('SELECT SUM(liked) FROM like_monitor WHERE user_id = ?', array($user_id));
+
+        return $count_likes;
+    }
+
+    public function countDislike($user_id)
+    {
+        $db = DB::conn();
+
+        $count_likes = $db->value('SELECT SUM(disliked) FROM like_monitor WHERE user_id = ?', array($user_id));
+
+        return $count_likes;
+    }
+
+    public function deleteLike($thread_id)
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+
+            $delete = $db->query('DELETE l FROM like_monitor l 
```

Comment:
You may use 3 delete queries po instead of 1 query since this is inside a transaction :bow: